pkg/util/metric/aggmetric/counter_test.go

```diff
@@ -72,3 +74,59 @@ func TestAggCounter(t *testing.T) {
 	testFile = "SQLCounter_post_eviction.txt"
 	echotest.Require(t, writePrometheusMetrics(t), datapathutils.TestDataPath(t, testFile))
 }
+
+func TestBoundedCounter(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	const cacheSize = 10
+	r := metric.NewRegistry()
+	writePrometheusMetrics := WritePrometheusMetricsFunc(r)
+
+	c := NewBoundedCounter(metric.Metadata{
+		Name: "foo_counter",
+	}, "database", "application_name")
+	c.mu.children = &UnorderedCacheWrapper{
+		cache: initialiseCacheStorageForTesting(),
+	}
+
+	r.AddMetric(c)
+
+	for i := 0; i < cacheSize+5; i++ {
+		c.Inc(1, "1", strconv.Itoa(i))
+	}
+
+	//wait more than cache eviction time to make sure that keys are not evicted based on only cache size.
+	time.Sleep(6 * time.Second)
+
+	testFile := "boundedCounter_pre_eviction.txt"
+	echotest.Require(t, writePrometheusMetrics(t), datapathutils.TestDataPath(t, testFile))
+
+	for i := 0 + cacheSize; i < cacheSize+5; i++ {
+		c.Inc(1, "2", strconv.Itoa(i))
+	}
+
+	testFile = "boundedCounter_post_eviction.txt"
+	echotest.Require(t, writePrometheusMetrics(t), datapathutils.TestDataPath(t, testFile))
+}
+
```

Comment:
Can we add a test to assert the eviction behaviour? Doesn't have to be a datadriven test. I think it can be done in a simple unit test